src/hooks/useDeck.js

```diff
@@ -18,35 +24,96 @@ function useDeck(initialState, numSlides, looping, animationsWhenGoingBack) {
         if (state.currentSlide + 1 === numSlides) {
           if (looping) {
             return action.type === 'NEXT_SLIDE_IMMEDIATE'
-              ? { ...initialState, immediate: true }
+              ? {
+                  ...initialState,
+                  immediate: true,
+                  reverseDirection: false,
+                  immediateElement: false
+                }
               : initialState;
           }
           return { ...state };
         }
         return action.type === 'NEXT_SLIDE_IMMEDIATE'
-          ? { currentSlide: state.currentSlide + 1, immediate: true }
-          : { currentSlide: state.currentSlide + 1 };
+          ? {
+              currentSlide: state.currentSlide + 1,
+              immediate: true,
+              currentSlideElement: 0,
+              immediateElement: false,
+              reverseDirection: false
+            }
+          : {
+              currentSlide: state.currentSlide + 1,
+              currentSlideElement: 0,
+              immediate: false,
+              immediateElement: false,
+              reverseDirection: false
+            };
       case 'PREV_SLIDE':
-        // If current slide is inital slide then if looping go
+        // If current slide is initial slide then if looping go
         // to last slide else stop
         if (state.currentSlide === initialState.currentSlide) {
           if (looping) {
             return {
               currentSlide: numSlides - 1,
-              immediate: animationsWhenGoingBack ? true : false
+              currentSlideElement: Math.max(slideElementMap[numSlides - 1], 0),
+              immediate: !!animationsWhenGoingBack,
+              reverseDirection: true
             };
           }
           return { ...state };
         }
         return {
           currentSlide: state.currentSlide - 1,
-          immediate: animationsWhenGoingBack ? true : false
+          currentSlideElement: Math.max(
+            slideElementMap[state.currentSlide - 1],
+            0
+          ),
+          immediate: !!animationsWhenGoingBack,
+          immediateElement: true,
+          reverseDirection: true
+        };
+      case 'NEXT_SLIDE_ELEMENT': {
```

Comment:
We add the element navigation to the deck reducer because it reduces the number of dispatches and keeps all navigation in a unified location. This will be key when we integrate URL-based routing. Aka `spectacle:3000/1/2` meaning slide 1, element 2. Tracking navigation of slides and their elements in separate reducers makes them individually smaller, but more challenging to keep data in sync.